### ISSUE TITLE
[Bug] metadata of segmentation extractor and imaging extractor interfaces are incompatible  

### ISSUE THREAD

**Issue by h-mayorquin** [catalystneuro/nwb-conversion-tools#537]
The metadata of `BaseImagingExtractorInterface` requires the metadata of `TwoPhotonInterface` to be an array:
https://github.com/catalystneuro/nwb-conversion-tools/blob/a03bf773bb9808255eeb893ba2d2cbb5211a5758/src/nwb_conversion_tools/datainterfaces/ophys/baseimagingextractorinterface.py#L36-L45

Whereas the one from `BaseSegmentationExtractorInterface` requires it to be an object

https://github.com/catalystneuro/nwb-conversion-tools/blob/a03bf773bb9808255eeb893ba2d2cbb5211a5758/src/nwb_conversion_tools/datainterfaces/ophys/basesegmentationextractorinterface.py#L31-L36

If you run that line in main you get the following:
```python
from nwb_conversion_tools.utils import get_schema_from_hdmf_class
from pynwb.ophys import TwoPhotonSeries

get_schema_from_hdmf_class(TwoPhotonSeries)["type"]
```

outputs `object`

What should we do? What should be aligned with what?

Here is an example that reproduces the error with a data class (I think we can use something of this kind to test general metadata combination)

```python

import unittest
from datetime import datetime

from nwb_conversion_tools import Suite2pSegmentationInterface, TiffImagingInterface
from nwb_conversion_tools import NWBConverter

from .setup_paths import OPHYS_DATA_PATH, OUTPUT_PATH


TiffImagingInterface_source_data = dict(
    file_path=str(OPHYS_DATA_PATH / "imaging_datasets" / "Tif" / "demoMovie.tif"), sampling_frequency=15.0
)
Suite2pSegmentationInterface_source_data = dict(file_path=str(OPHYS_DATA_PATH / "segmentation_datasets" / "suite2p"))

interface_to_source_data_map = dict(
    TiffImagingInterface=TiffImagingInterface_source_data,
    Suite2pSegmentationInterface=Suite2pSegmentationInterface_source_data,
)

interface_to_class_map = dict(
    TiffImagingInterface=TiffImagingInterface,
    Suite2pSegmentationInterface=Suite2pSegmentationInterface,
)

interfaces_to_combine = ["TiffImagingInterface", "Suite2pSegmentationInterface"]


class TestConverter(NWBConverter):
    data_interface_classes = {interface: interface_to_class_map[interface] for interface in interfaces_to_combine}


source_data = {interface: interface_to_source_data_map[interface] for interface in interfaces_to_combine}
converter = TestConverter(source_data=source_data)
metadata = converter.get_metadata()
assert converter.validate_metadata(metadata=metadata)
```

The traceback

```python
Traceback (most recent call last):
  File "/home/heberto/development/nwb-conversion-tools/tests/test_on_data/test_metadata_combinations.py", line 37, in <module>
    converter.validate_metadata(metadata=metadata)
  File "/home/heberto/development/nwb-conversion-tools/src/nwb_conversion_tools/nwbconverter.py", line 113, in validate_metadata
    validate(instance=decoded_metadata, schema=self.get_metadata_schema())
  File "/home/heberto/miniconda3/envs/nwb_conversion_testing_env/lib/python3.9/site-packages/jsonschema/validators.py", line 934, in validate
    raise error
jsonschema.exceptions.ValidationError: [{'name': 'TwoPhotonSeries', 'description': 'no description', 'comments': 'Generalized from RoiInterface', 'unit': 'n.a.', 'dimension': [60, 80], 'rate': 15.0, 'imaging_plane': 'ImagingPlane'}] is not of type 'object'

Failed validating 'type' in schema['properties']['Ophys']['properties']['TwoPhotonSeries']:
    {'additionalProperties': False,
     'default': [{'comments': 'Generalized from RoiInterface',
                  'description': 'no description',
                  'dimension': [60, 80],
                  'imaging_plane': 'ImagingPlane',
                  'name': 'TwoPhotonSeries',
                  'rate': 15.0,
                  'unit': 'n.a.'}],
     'items': {'$ref': '#/properties/Ophys/properties/definitions/TwoPhotonSeries'},
     'minItems': 1,
     'properties': {'comments': {'default': 'no comments',
                                 'description': 'Human-readable comments '
                                                'about this TimeSeries '
                                                'dataset',
                                 'type': 'string'},
                    'control': {'description': 'Numerical labels that '
                                               'apply to each element in '
                                               'data',
                                'type': 'array'},
                    'control_description': {'description': 'Description of '
                                                           'each control '
                                                           'value',
                                            'type': 'array'},
                    'conversion': {'default': 1.0,
                                   'description': 'Scalar to multiply each '
                                                  'element in data to '
                                                  'convert it to the '
                                                  'specified unit',
                                   'type': 'number'},
                    'description': {'default': 'no description',
                                    'description': 'Description of this '
                                                   'TimeSeries dataset',
                                    'type': 'string'},
                    'device': {'description': 'Device used to capture the '
                                              'images/video.',
                               'target': 'pynwb.device.Device',
                               'type': 'string'},
                    'dimension': {'description': 'Number of pixels on x, '
                                                 'y, (and z) axes.',
                                  'type': 'array'},
                    'field_of_view': {'description': 'Width, height and '
                                                     'depth of image, or '
                                                     'imaged area '
                                                     '(meters).',
                                      'type': 'array'},
                    'format': {'description': 'Format of image. Three '
                                              'types: 1) Image format; '
                                              'tiff, png, jpg, etc. 2) '
                                              'external 3) raw.',
                               'type': 'string'},
                    'imaging_plane': {'description': 'Imaging plane '
                                                     'class/pointer.',
                                      'target': 'pynwb.ophys.ImagingPlane',
                                      'type': 'string'},
                    'name': {'description': 'The name of this TimeSeries '
                                            'dataset',
                             'type': 'string'},
                    'offset': {'default': 0.0,
                               'description': 'Scalar to add to each '
                                              'element in the data scaled '
                                              "by 'conversion' to finish "
                                              'converting it to the '
                                              'specified unit.',
                               'type': 'number'},
                    'pmt_gain': {'description': 'Photomultiplier gain.',
                                 'type': 'number'},
                    'rate': {'description': 'Sampling rate in Hz',
                             'type': 'number'},
                    'resolution': {'default': -1.0,
                                   'description': 'The smallest meaningful '
                                                  'difference (in '
                                                  'specified unit) between '
                                                  'values in data',
                                   'type': 'number'},
                    'scan_line_rate': {'description': 'Lines imaged per '
                                                      'second. This is '
                                                      'also stored in '
                                                      '/general/optophysiology '
                                                      'but is kept here as '
                                                      'it is useful '
                                                      'information for '
                                                      'analysis, and so '
                                                      'good to be stored '
                                                      'w/ the actual data.',
                                       'type': 'number'},
                    'starting_frame': {'default': [0],
                                       'description': 'Each entry is the '
                                                      'frame number in the '
                                                      'corresponding '
                                                      'external_file '
                                                      'variable. This '
                                                      'serves as an index '
                                                      'to what frames each '
                                                      'file contains. If '
                                                      'external_file is '
                                                      'not provided, then '
                                                      'this value will be '
                                                      'None',
                                       'type': 'array'},
                    'starting_time': {'description': 'The timestamp of the '
                                                     'first sample',
                                      'type': 'number'},
                    'unit': {'description': 'The unit of measurement of '
                                            'the image data, e.g., values '
                                            'between 0 and 255. Required '
                                            'when data is specified. If '
                                            'unit (and data) are not '
                                            'specified, then unit will be '
                                            'set to "unknown".',
                             'type': 'string'}},
     'required': ['name', 'imaging_plane'],
     'tag': 'pynwb.ophys.TwoPhotonSeries',
     'type': 'object'}

On instance['Ophys']['TwoPhotonSeries']:
    [{'comments': 'Generalized from RoiInterface',
      'description': 'no description',
      'dimension': [60, 80],
      'imaging_plane': 'ImagingPlane',
      'name': 'TwoPhotonSeries',
      'rate': 15.0,
      'unit': 'n.a.'}]

```

**Comment by h-mayorquin** [comment]
OK, I think I found a workaround. See #538 

However, I am wondering the following:
Right now, we have objects like `Device` and `ImagingPlane` at the top of the `Ophys` schema, This objects though usually are bound to other objects when we build the `nwb_file` (e.g. `ImagingPlane` for example is a property of `TwoPhotonSeries` in the imaging interface and ends up there):

https://github.com/catalystneuro/nwb-conversion-tools/blob/a03bf773bb9808255eeb893ba2d2cbb5211a5758/src/nwb_conversion_tools/datainterfaces/ophys/baseimagingextractorinterface.py#L36-L45

There is a similar situation with `Device` and `ImagingPlane` at the top level of `Ophys` in the schema for the `BaseSegmentationExtractorInterface`:

https://github.com/catalystneuro/nwb-conversion-tools/blob/a03bf773bb9808255eeb893ba2d2cbb5211a5758/src/nwb_conversion_tools/datainterfaces/ophys/basesegmentationextractorinterface.py#L28-L37

**My question is**, would it not be better to have this metadata nested to the level that they actually belong to?

For example, for the `BaseImagingExtractorInterface` would it not make more sense to have `ImagingPlane` as a property of the `TwoPhotonSeries` at the metadata level? That is the level that it occupies in the actual schema and we eventually send the object there with the `add_two_photon_series` function. 

The only reason that I can see to keep these objects like `Device` and `ImagingPlane` at the top level is that maybe is easier to link themacross modalities (for example if the segmentation extractor and the imaging extractor share an `ImagingPlane` or a `Device`). In that case, I think that we can simple link the objects with the name if the device already exists in the `nwb_file`. 

Similar concenrs applie for `Device` in both classess and to `TwoPhotonSeries` in the `BaseSegmentationExtractorInterface` (where it should be nested in `ImageSegmentation`).  **I feel that setting them at the right (nested) level avoids a type of ambiguity when, for example, more than one Device or ImagingPlane is used and also allows for better control of the properties of the schema.**

Maybe I am missing something?

TL;DR We should have the metadata properly nested in the place that they occupy in the schema and link by name if we need to.

**Comment by CodyCBakerPhD** [comment]
Yep, general workaround to this is the add a line or two to the `get_metadata_schema` method to hard-code the intended value. The automatic methods for parsing `hdmf` docvals is 'nice' but not always 100% accurate.

I do think though that things like this would be even easier to fix if the entire metadata structure was written and centralized in a `.yml` file where it can be read and used by the interfaces with ease.
